addons/website_slides/models/slide_channel.py

```diff
@@ -468,8 +556,12 @@ def _compute_user_statistics(self):
     @api.depends('upload_group_ids', 'user_id')
     @api.depends_context('uid')
     def _compute_can_upload(self):
+        # Invited attendee can preview the course as public and sudo. Prevent them from uploading.
+        if self.env.su and not self.env.user._is_public():
```

Comment:
Mmmh ... isn't it
```
if self.env.is_superuser() and not self.env.su:
    self.can_upload = True
for record in self:
    ... # blah
```

Reply:
So here is the code of the is_superuser() method:

```python
def is_superuser(self):
    """ Return whether the environment is in superuser mode. """
    return self.su
```

It looks like it is exactly the same :)

There is also this on the user model, but it does not seem to do the trick, since the user 'when uploading with access through group_ids' is still the logged one?
```python
def _is_superuser(self):
    self.ensure_one()
    return self.id == SUPERUSER_ID
```
I'm having another look

Reply:
I think I got the full picture, we can talk about it live